<a href="https://colab.research.google.com/github/mariajosevezga24-hash/PROGRAMACION/blob/CLASES/Parte_II_Y_III.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Nombre : Maria Jose Caicedo Vezga  

Codigo : 2231946

INSTRUCCIÓN 1

Genere un arreglo de NumPy con 50 números enteros aleatorios entre 1XY y 19Z,
donde XY corresponde a los dos últimos dígitos de la TI o CC y Z al último dígito de la edad.
Utilice el siguiente código:
np.random.seed(0)
datos = np.random.randint(1XY, 19Z, 50)
1. Implemente una filtración sobre el elemento datos e imprima el elemento con los valores
resultantes.
2. Escale cada uno de los valores de datos utilizando una función lambda:
𝑥􀯜 − 𝑥̅
𝑆􀯫
Para 𝑥̅ la media aritmética y 𝑆􀯫 la desviación estándar.
3. Implemente el método .describe() e interprete los resultados.
4. Convierta el elemento datos en un DataFrame usando pandas. Imprima el elemento.
5. Exporte los resultados a un archivo CSV llamado resultados.csv
Instrucción 2: Construya un DataFrame en pandas con información de su

In [40]:
import numpy as np
import pandas as pd
np.random.seed(0)
datos = np.random.randint(173, 192, 50)


In [89]:
x =  np.random.randint(173, 192, 50)
x = pd.DataFrame(x)
normalizar = lambda x: (x - x.mean()) / x.std()
resultado = normalizar(x)
print(resultado)

           0
0  -1.293868
1  -1.293868
2  -0.239084
3   0.112510
4  -0.942273
5  -0.414882
6   0.464105
7   0.991497
8   1.694686
9  -1.469665
10  0.991497
11 -0.942273
12  0.639902
13  0.288308
14  0.464105
15 -0.766476
16 -0.414882
17 -0.766476
18  1.167294
19 -0.942273
20  0.639902
21 -0.766476
22 -0.063287
23  0.991497
24  1.167294
25 -0.942273
26  1.167294
27  0.815699
28  1.343091
29  1.518889
30 -0.590679
31  0.112510
32 -0.942273
33 -1.469665
34 -0.590679
35 -1.469665
36  1.518889
37  1.694686
38 -0.766476
39 -1.118071
40  1.343091
41 -0.942273
42 -1.118071
43  0.288308
44  0.815699
45  1.343091
46 -0.239084
47  0.112510
48 -1.469665
49  0.288308


In [ ]:
from google.colab import files
import csv
datos_procesados =

In [113]:
import pandas as pd
resultados = pd.DataFrame(np.random.randint(173, 192, 50))
resultados

,0
0,173
1,186
2,186
3,178
4,188
5,180
6,191
7,177
8,187
9,185


In [ ]:
import csv
resultados
with open("datos_procesados.csv", "w", newline="") as f:
  writer = csv.writer(f)
  writer.writerows(datos_procesados)

In [81]:
x.describe()

,0
count,50.000000
mean,181.520000
std,5.768458
min,173.000000
25%,176.250000
50%,180.000000
75%,185.750000
max,191.000000


Nuestra muestra es de 50 datos tiene media 181.52 , desviacion estandar de 5.76 sus cuantiles del 25% , 50% y 75% correspondientemente son : 176.25, 180 y 185.75

INTRUCCIÓN II

Construya un DataFrame en pandas con información de su elección. El
DataFrame debe cumplir con: (1) al menos 4 columnas (por ejemplo: nombre, edad, ingreso,
ciudad), (2) al menos 5 filas (registros) y (3) incluir datos numéricos y categóricos (o variable
cualitativa)
1. Muestre las primeras filas usando método .head()
2. Seleccione una columna específica de su elección.
3. Seleccione un conjunto de filas dadas condiciones específicas sobre valores de la columna.
4. Cree una nueva columna que represente una transformación de una variable.
5. Agrupe los datos por una variable categórica usando el método .groupby(). Por ejemplo:
 Agrupar por ciudad y calcular ingreso promedio
grupo = df.groupby("Ciudad")["Ingreso"].mean()
6. Exporte el DataFrame final a un archivo CSV llamado datos_procesados.csv

In [15]:
import pandas as pd
df = pd.DataFrame ({
    "Nombre":["Ana", "Luis", "Carlos", "Ian"],
    "Ciudad":["Bogota", "Medellin", "Cali", "Barranquilla"],
    "Edad":[22, 30, 35, 28],
    "Salario":[1500, 2000, 3000, 2500]
})
df

,Nombre,Ciudad,Edad,Salario
0,Ana,Bogota,22,1500
1,Luis,Medellin,30,2000
2,Carlos,Cali,35,3000
3,Ian,Barranquilla,28,2500


In [16]:
df.head(2) # Primeras filas

,Nombre,Ciudad,Edad,Salario
0,Ana,Bogota,22,1500
1,Luis,Medellin,30,2000


In [17]:
df["Edad"] # Para imprimir la columna edad

,Edad
0,22
1,30
2,35
3,28


In [18]:
df["Edad"].apply(lambda x: x+5) #fijo funciones lambda y Suma 5 a la columna edad

,Edad
0,27
1,35
2,40
3,33


In [21]:
grupo = df.groupby("Ciudad")["Salario"].mean()
grupo

,Salario
Ciudad,
Barranquilla,2500.0
Bogota,1500.0
Cali,3000.0
Medellin,2000.0


In [29]:
import csv
datos_procesados = [
    ["Nombre", "Ciudad", "Edad", "Salario"],
    ["Ana", "Bogota", 22, 1500],
    ["Luis", "Medellin", 30, 2000],
    ["Carlos", "Cali", 35, 3000],
    ["Ian", "Barranquilla", 28, 2500]
]
with open("datos_procesados.csv", "w", newline="") as f:
  writer = csv.writer(f)
  writer.writerows(datos_procesados)

In [31]:
with open("datos_procesados.csv", "r") as f:
    print(f.read())

Nombre,Ciudad,Edad,Salario
Ana,Bogota,22,1500
Luis,Medellin,30,2000
Carlos,Cali,35,3000
Ian,Barranquilla,28,2500



In [36]:
from google.colab import files
files.download("datos_procesados.csv")

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

PARTE III

In [39]:

import numpy as np
import pandas as pd

np.random.seed(123)

grupo_A = np.random.normal(loc=7.2, scale=0.4, size=54)
grupo_B = np.random.gamma(shape=306, scale=0.0229, size=54)

df = pd.DataFrame({
    "A": grupo_A,
    "B": grupo_B
})

In [41]:
df.describe()

,A,B
count,54.000000,54.000000
mean,7.194640,7.101657
std,0.482729,0.386488
min,6.080564,6.376485
25%,6.860937,6.851536
50%,7.137259,7.107928
75%,7.568949,7.398087
max,8.156946,8.092430


In [48]:
from scipy.stats import shapiro # Prueba shapiro para la normal
shapiro(df["A"])

ShapiroResult(statistic=np.float64(0.9818971037941049), pvalue=np.float64(0.5848876244931867))

In [49]:
from scipy.stats import shapiro # Prueba shapiro para la Gamma
shapiro(df["B"])

ShapiroResult(statistic=np.float64(0.9838241504529996), pvalue=np.float64(0.6756374124038879))

No se rechaza la hpótesis de normalidad con una confianza del 95%

In [42]:
from scipy import stats
t_stat, p_val = stats.ttest_ind(df["A"], df["B"])

print("Estadístico t:", t_stat)
print("Valor p:", p_val)

Estadístico t: 1.104938620482421
Valor p: 0.27168792466612407


No se rechaza con una confianza del 95% que los puntajes de ambas encuestas estan es igual

In [51]:
stats.kendalltau(df["A"], df["B"])

SignificanceResult(statistic=np.float64(-0.09853249475890984), pvalue=np.float64(0.2928377927852378))

No se rechaza con una confianza del 95% que los puntajes de ambas encuestas no están correlacionados pues p>0.05

In [54]:
stats.spearmanr(df["A"], df["B"])

SignificanceResult(statistic=np.float64(-0.12368972746331236), pvalue=np.float64(0.37288046234302963))

No se rechaza con una confianza del 95% que los puntajes de ambas encuestas no están correlacionados

In [59]:

import os
os.makedirs("comparación", exist_ok=True)

In [60]:
%%writefile comparación/operaciones.py

Writing comparación/operaciones.py


In [62]:
%%writefile comparación/pruebat.py

Writing comparación/pruebat.py


In [63]:
%%writefile comparación/cor.py

Writing comparación/cor.py


In [65]:
df = pd.DataFrame({
"Estudiante": range(1, 11),
"Prueba_A": [7.1, 7.5, 6.8, 7.3, 7.0, 7.4, 6.9, 7.2, 7.6, 7.1],
"Prueba_B": [6.9, 7.2, 7.0, 7.1, 6.8, 7.3, 7.0, 7.1, 7.4, 7.0]
})